### PR TITLE
feat(DCS): add dcs custom template resource

### PR DESCRIPTION
**Pull request by houpeng80**
**What this PR does / why we need it**:
  add dcs custom template resource
**Which issue this PR fixes**:
*(optional, in `fixes #<issue number>(, fixes #<issue_number>, ...)` format, will close that issue when PR gets merged)*
fixes #xxx

**Special notes for your reviewer**:

**Release note**:

```release-note
  add dcs custom template resource
```

## PR Checklist

* [ ] Tests added/passed.
* [ ] Documentation updated.
* [ ] Schema updated.

## Acceptance Steps Performed

```
make testacc TEST=./huaweicloud/services/acceptance/dcs/ TESTARGS='-run TestAccCustomTemplate_basic'
==> Checking that code complies with gofmt requirements...
TF_ACC=1 go test ./huaweicloud/services/acceptance/dcs/ -v -run TestAccCustomTemplate_basic -timeout 360m -parallel 4
=== RUN   TestAccCustomTemplate_basic
=== PAUSE TestAccCustomTemplate_basic
=== CONT  TestAccCustomTemplate_basic
--- PASS: TestAccCustomTemplate_basic (14.37s)
PASS
ok      github.com/huaweicloud/terraform-provider-huaweicloud/huaweicloud/services/acceptance/dcs       14.412s
```

### CODE REVIEW

File: docs/resources/dcs_custom_template.md [hunk 1]
```diff
@@ -0,0 +1,119 @@
...
+```
+
+Note that the imported state may not be identical to your resource definition, due to some attributes missing from the
+API response, security or some other reason. The missing attributes include: `template_id`, `source_type`, `params`. It
+is generally recommended running `terraform plan` after importing a custom template. You can then decide if changes
+should be applied to the custom template, or the resource definition should be updated to align with the DCS custom
+template. Also you can ignore changes as below.
+
+```hcl
+resource "huaweicloud_organizations_account_invite" "test" {
```
Comment (deer-hang) [inline docs/resources/dcs_custom_template.md]:
```suggestion
resource "huaweicloud_dcs_custom_template" "test" {
```

File: docs/resources/dcs_custom_template.md [hunk 2]
```diff
@@ -0,0 +1,119 @@
...
+
+## Attribute Reference
+
+In addition to all arguments above, the following attributes are exported:
+
+* `id` - The resource ID.
+
+* `type` - Indicates the type of the template. The value can be **sys**, **user**.
+
+* `engine` - Indicates the cache engine. The value can be **Redis**.
```
Comment (deer-hang) [inline docs/resources/dcs_custom_template.md]:
```suggestion
* `engine` - Indicates the cache engine. Currently, only **Redis** supported.
```

File: huaweicloud/services/dcs/resource_huaweicloud_dcs_custom_template.go [hunk 3]
```diff
@@ -0,0 +1,432 @@
...
+			"params": {
+				Type:        schema.TypeSet,
+				Elem:        customTemplateParamSchema(),
+				Required:    true,
+				Description: `Specifies the list of the template params.`,
+			},
+			"description": {
+				Type:        schema.TypeString,
+				Optional:    true,
+				Computed:    true,
```
Comment (ShiChangkuo) [inline huaweicloud/services/dcs/resource_huaweicloud_dcs_custom_template.go]: `Computed` should be deleted

File: huaweicloud/services/dcs/resource_huaweicloud_dcs_custom_template.go [hunk 4]
```diff
@@ -0,0 +1,432 @@
...
+			return diag.Errorf("error updating DCS custom template: %s", err)
+		}
+	}
+	return resourceCustomTemplateRead(ctx, d, meta)
+}
+
+func buildUpdateCustomTemplateBodyParams(d *schema.ResourceData) map[string]interface{} {
+	bodyParams := map[string]interface{}{
+		"name":        utils.ValueIngoreEmpty(d.Get("name")),
+		"description": utils.ValueIngoreEmpty(d.Get("description")),
```
Comment (ShiChangkuo) [inline huaweicloud/services/dcs/resource_huaweicloud_dcs_custom_template.go:378]: support clear description

```suggestion
		"description": d.Get("description"),
```

**Pull request merged**